Webpack build error in v8.4.1 for `react-dnd-touch-backend`

### React Query Builder version

V8.4.1 

### Platform

node V18.19.1

### Description

when use v8.4.1 build, Tell me there is an error, Downgrading to V8.3 is normal.erro message

### Reproduction

`
Error: ERROR in ./node_modules/.pnpm/@react-querybuilder+dnd@8.4.1_react-dnd-html5-backend@16.0.1_react-dnd@16.0.1_@types+ho_61e007f03c6d1c84a9d4fd6ff23eb7e5/node_modules/@react-querybuilder/dnd/dist/react-querybuilder_dnd.mjs 1285:18-51
[01:34:06] Module not found: Error: Can't resolve 'react-dnd-touch-backend' in '/root/workspace/hm-web_i8Qq/node_modules/.pnpm/@react-querybuilder+dnd@8.4.1_react-dnd-html5-backend@16.0.1_react-dnd@16.0.1_@types+ho_61e007f03c6d1c84a9d4fd6ff23eb7e5/node_modules/@react-querybuilder/dnd/dist'


[01:34:06] webpack compiled with 1 error
[01:34:06]     at handler (/root/workspace/hm-web_i8Qq/node_modules/.pnpm/@umijs+bundler-webpack@4.4.6_type-fest@1.4.0_typescript@5.8.2_webpack@5.98.0/node_modules/@umijs/bundler-webpack/dist/build.js:82:79)
[01:34:06]     at finalCallback (/root/workspace/hm-web_i8Qq/node_modules/.pnpm/@umijs+bundler-webpack@4.4.6_type-fest@1.4.0_typescript@5.8.2_webpack@5.98.0/node_modules/@umijs/bundler-webpack/compiled/webpack/index.js:62945:32)
[01:34:06]     at /root/workspace/hm-web_i8Qq/node_modules/.pnpm/@umijs+bundler-webpack@4.4.6_type-fest@1.4.0_typescript@5.8.2_webpack@5.98.0/node_modules/@umijs/bundler-webpack/compiled/webpack/index.js:62962:13
[01:34:06]     at Hook.eval [as callAsync] (eval at create (/root/workspace/hm-web_i8Qq/node_modules/.pnpm/@umijs+bundler-utils@4.4.6/node_modules/@umijs/bundler-utils/compiled/tapable/index.js:1:7682), <anonymous>:35:1)
[01:34:06]     at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/root/workspace/hm-web_i8Qq/node_modules/.pnpm/@umijs+bundler-utils@4.4.6/node_modules/@umijs/bundler-utils/compiled/tapable/index.js:1:4851)
[01:34:06]     at onCompiled (/root/workspace/hm-web_i8Qq/node_modules/.pnpm/@umijs+bundler-webpack@4.4.6_type-fest@1.4.0_typescript@5.8.2_webpack@5.98.0/node_modules/@umijs/bundler-webpack/compiled/webpack/index.js:62960:21)
[01:34:06]     at /root/workspace/hm-web_i8Qq/node_modules/.pnpm/@umijs+bundler-webpack@4.4.6_type-fest@1.4.0_typescript@5.8.2_webpack@5.98.0/node_modules/@umijs/bundler-webpack/compiled/webpack/index.js:63704:17
[01:34:06]     at _next0 (eval at create (/root/workspace/hm-web_i8Qq/node_modules/.pnpm/@umijs+bundler-utils@4.4.6/node_modules/@umijs/bundler-utils/compiled/tapable/index.js:1:7682), <anonymous>:16:1)
[01:34:06]     at eval (eval at create (/root/workspace/hm-web_i8Qq/node_modules/.pnpm/@umijs+bundler-utils@4.4.6/node_modules/@umijs/bundler-utils/compiled/tapable/index.js:1:7682), <anonymous>:26:1)
[01:34:06]     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)`

### Expected behavior

_No response_

### Additional information

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,6 +13,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   - The `translations` prop object has a new property `values` that accepts `placeholderName`, `placeholderLabel`, and `placeholderGroupLabel` properties ([documentation](https://react-querybuilder.js.org/docs/api/querybuilder#translations)). These translatable strings set the default values and labels when `autoSelectValue` is set to `false`.
   - A corresponding option was also added to [`formatQuery`](https://react-querybuilder.js.org/docs/utils/export#placeholder-values), which will now ignore rules where the `value` matches the placeholder value as long as `placeholderValueName` is defined (this behavior differs from `placeholderFieldName` and `placeholderOperatorName`, which do not need to be defined).
 
+### Fixed
+
+- [#877] Dynamic imports of `react-dnd` and related packages are now obfuscated to discourage bundlers like webpack from pre-processing those imports.
+
 ## [v8.4.1] - 2025-03-24
 
 ### Fixed
@@ -1886,6 +1890,7 @@ Maintenance release focused on converting to a monorepo with Vite driving the bu
 [#860]: https://github.com/react-querybuilder/react-querybuilder/pull/860
 [#868]: https://github.com/react-querybuilder/react-querybuilder/pull/868
 [#873]: https://github.com/react-querybuilder/react-querybuilder/pull/873
+[#877]: https://github.com/react-querybuilder/react-querybuilder/pull/877
 
 <!-- #endregion -->
 

--- a/packages/dnd/src/QueryBuilderDnD.tsx
+++ b/packages/dnd/src/QueryBuilderDnD.tsx
@@ -195,11 +195,11 @@ export const useReactDnD = (dndParam?: DndProp): UseReactDnD | null => {
     let didCancel = false;
 
     const getDnD = async () => {
-      const [reactDnD, reactDndHTML5Be, reactDndTouchBe] = await Promise.all([
-        import('react-dnd').catch(() => null),
-        import('react-dnd-html5-backend').catch(() => null),
-        import('react-dnd-touch-backend').catch(() => null),
-      ]);
+      const [reactDnD, reactDndHTML5Be, reactDndTouchBe] = await Promise.all(
+        ['', '-html5-backend', '-touch-backend'].map(pn =>
+          import(`react-dnd${pn}`).catch(() => null)
+        )
+      );
 
       // istanbul ignore else
       if (!didCancel) {

